includes/html/pages/addhost.inc.php

```diff
@@ -147,19 +147,28 @@
           <label for="snmpver" class="col-sm-3 control-label">SNMP Version</label>
           <div class="col-sm-3">
             <select name="snmpver" id="snmpver" class="form-control input-sm" onChange="changeForm();">
-              <option value="v1">v1</option>
-              <option value="v2c" selected>v2c</option>
-              <option value="v3">v3</option>
-            </select>
+                <?php
+                $snmpver_list = [ "v1", "v2c", "v3" ];
+                $snmpver_pref = Config::get('snmp.version.0', 'v2c');
+                foreach ($snmpver_list as $snmpver_item) {
+                    $selected = '';
+                    $snmpver_item == $snmpver_pref ? $selected = 'selected' : '';
+                    echo "<option value=\"" . $snmpver_item ."\"" . $selected . ">" . $snmpver_item. "</option>";
+                }
+                ?>
+	    </select>
           </div>
           <div class="col-sm-3">
-            <input type="text" name="port" placeholder="port" class="form-control input-sm">
+            <input type="text" name="port" placeholder="port" class="form-control input-sm" value="<?php echo Config::get('snmp.port', 161) ?>">
```

Comment:
whoops unescaped output, this adds a security vulnerabilty.

Reply:
I'll push a htmlspecialchars fix soon